Index: src/app/application.cpp
===================================================================
--- src.orig/app/application.cpp	2022-11-26 16:16:23.000000000 -0500
+++ src/app/application.cpp	2023-01-02 14:18:17.144113689 -0500
@@ -706,7 +706,7 @@
 #ifndef DISABLE_GUI
     UIThemeManager::initInstance();
 
-    m_desktopIntegration = new DesktopIntegration(this);
+    m_desktopIntegration = new DesktopIntegration;
     m_desktopIntegration->setToolTip(tr("Loading torrents..."));
 #ifndef Q_OS_MACOS
     auto *desktopIntegrationMenu = new QMenu;
@@ -1201,6 +1201,7 @@
         ::ShutdownBlockReasonDestroy(reinterpret_cast<HWND>(m_window->effectiveWinId()));
 #endif // Q_OS_WIN
         delete m_window;
+        delete m_desktopIntegration;
         UIThemeManager::freeInstance();
     }
 #endif // DISABLE_GUI
